unit_tests/src/test_metadata.cpp

```diff
@@ -0,0 +1,285 @@
+#include "catch.hpp"
+
+#include "getRodsEnv.h"
+#include "rodsClient.h"
+#include "rcConnect.h"
+#include "connection_pool.hpp"
+#include "filesystem.hpp"
+#include "metadata.hpp"
+
+#include "irods_at_scope_exit.hpp"
+#include "dstream.hpp"
+#include "transport/default_transport.hpp"
+
+#include <vector>
+#include <string>
+
+namespace ix  = irods::experimental;
+namespace ixm = ix::metadata;
+using entity_type = ix::entity::type;
+
+TEST_CASE("metadata")
+{
+    load_client_api_plugins();
+
+    rodsEnv env;
+    REQUIRE(getRodsEnv(&env) == 0);
+
+    const int cp_size = 1;
+    const int cp_refresh_time = 600;
+
+    irods::connection_pool conn_pool{cp_size,
+                                     env.rodsHost,
+                                     env.rodsPort,
+                                     env.rodsUserName,
+                                     env.rodsZone,
+                                     cp_refresh_time};
+
+    auto conn = conn_pool.get_connection();
```

Comment:
this can be replaced with `irods::experimental::client_connection`, but this is also fine as is